### PR TITLE
Implemented env[] keyword and tests.

### PR DESCRIPTION
**Pull request by mdmcconnell**
For consideration:  The env[] keyword allows an expression to access identifiers in env directly, with any name desired, including numbers, spaces, special characters, etc.  Syntax env['string identifier'] and env["string identifier"] are allowable.

### CODE REVIEW

**Comment by antonmedv** [comment]
Will review this.

File: parser/lexer/state.go [hunk 1]
```diff
@@ -128,6 +135,43 @@ loop:
 	return root
 }
 
+// Process what comes after the env keyword, which must be any
+// set of characters or a string literal, enclosed in square brackets
+func literalIdentifier(l *lexer) stateFn {
```
Comment (antonmedv) [inline parser/lexer/state.go:140]: Let's not introduce new syntax. We can just require to use strings:
```
env['foo bar']
```

**Review by antonmedv — CHANGES_REQUESTED** [review]

File: parser/lexer/state.go (hunk 1 above)
Comment (mdmcconnell) [reply parser/lexer/state.go:140]: Fair enough - it's not worthwhile to have new syntax for the sake of a two character savings.

**Comment by mdmcconnell** [comment]
Bump - this is working well, if you'd like to pull the code.

File: parser/lexer/state.go [hunk 2]
```diff
@@ -119,6 +124,8 @@ loop:
 				return not
 			case "in", "or", "and", "matches", "contains", "startsWith", "endsWith":
 				l.emit(Operator)
+			case "env":
+				return literalIdentifier
```
Comment (antonmedv) [inline parser/lexer/state.go:128]: Let’s implement this without altering lexer.

**Review by antonmedv — CHANGES_REQUESTED** [review]

File: parser/lexer/state.go (hunk 2 above)
Comment (mdmcconnell) [reply parser/lexer/state.go:128]: I feel that handling tokens within lexer is a clear, simple way to accomplish this.  Is the concern that too much interpretation of "meaning" of expressions moves into lexer?  Other keywords in, matches, contrains, startWith, endsWith are also interpreted within lexer.  I will look at an alternative.  Perhaps env can be a normal indexed identifier that parser will autmatically point at "env".  I suspect that will cause copmlications worrying about the type of the underlying value that the index returns; we'll see.

Comment (antonmedv) [reply parser/lexer/state.go:128]: env is just a variable.

**Comment by mdmcconnell** [comment]
Have moved env from lexer to parser.

**Comment by mdmcconnell** [comment]
Added about env note in Language Definition.

**Pull request closed without merging**